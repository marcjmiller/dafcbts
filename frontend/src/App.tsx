import React from 'react';
import {CssBaseline} from "@material-ui/core";

const App = () => {
  return (
    <React.Fragment>
      <CssBaseline />
      <div className="App">
        <header className="App-header">
          Hello World
        </header>
      </div>
    </React.Fragment>
  );
};

export default App;