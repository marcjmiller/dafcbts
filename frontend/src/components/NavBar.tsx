import { AppBar, Button, createStyles, IconButton, Theme, Toolbar, Typography } from '@material-ui/core';
import MenuIcon from '@material-ui/icons/Menu';
import Brightness7Icon from '@material-ui/icons/Brightness7';
import Brightness4Icon from '@material-ui/icons/Brightness4';
import { makeStyles } from '@material-ui/core/styles';
import * as React from 'react';
import { ThemeTypes } from '../types';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      flexGrow: 1,
    },
    menuButton: {
      marginRight: theme.spacing(2),
    },
    title: {
      flexGrow: 1,
    },
  }),
);

interface OwnProps {
  themeType: string;
  toggleThemeType: () => void;
}

export const NavBar: React.FC<OwnProps> = (props) => {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      <AppBar position="static">
        <Toolbar>
          <IconButton edge="start" className={classes.menuButton} color="inherit" aria-label="menu">
            <MenuIcon/>
          </IconButton>
          <Typography variant="h6" className={classes.title}>
            DumbAFCbts
          </Typography>
          <Button color="inherit">Login</Button>
          <IconButton color="inherit" onClick={props.toggleThemeType}>
            {
              props.themeType === ThemeTypes.DARK ?
                <Brightness7Icon/>
                :
                <Brightness4Icon/>
            }
          </IconButton>

        </Toolbar>
      </AppBar>
    </div>
  );
}