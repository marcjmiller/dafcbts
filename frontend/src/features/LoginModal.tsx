import React from 'react';
import { useStyles } from '../resources/theme';
import { useDispatch } from 'react-redux';
import { Button, Modal, TextField } from '@material-ui/core';
import { resetSignIn, signInSuccess } from '../store/reducer/slices/authSlice';
import classNames from 'classnames';


interface MyProps {
  display: boolean;
}

const LoginModal: React.FC<MyProps> = (props) => {
  const classes = useStyles();
  const dispatch = useDispatch();

  return (
    <Modal
      open={props.display}
      onClose={() => dispatch(resetSignIn())}
      style={{ top: '50%', left: '50%' }}
    >
      <div className={classNames(classes.modalBody, classes.paper, 'login--modal')}>
        <div className={'modal-text'}>
          <TextField id="username-email--input" label="Username or Email" variant="outlined"/>
        </div>
        <div className={'modal-text'}>
          <TextField id="password--input" label="Password" type="password" variant="outlined"/>
        </div>
        <Button variant="contained" color="primary"
                onClick={() => dispatch(signInSuccess({ userName: 'Billy', token: 'is logged in' }))}>
          Login
        </Button>
      </div>
    </Modal>
  );
};

export default LoginModal;