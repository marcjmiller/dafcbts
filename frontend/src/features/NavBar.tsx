import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { RootState, ThemeType } from '../resources/types';
import { useStyles } from '../resources/theme';
import { AuthStep, signOut, signInStart } from '../store/reducer/slices/authSlice';
import { AppBar, Button, IconButton, Toolbar, Typography } from '@material-ui/core';
import MenuIcon from '../resources/icons/MenuIcon';
import LightbulbIcon from '../resources/icons/LightbulbIcon';

interface MyProps {
  themeType: ThemeType;
  toggleTheme: () => void;
}

export const NavBar: React.FC<MyProps> = (props) => {
  const { authStep, user } = useSelector(
    (state: RootState) => state.auth,
  );

  const classes = useStyles();
  const dispatch = useDispatch();

  const isLoggedIn = authStep === AuthStep.LOGGED_IN;

  return (
    <div className={classes.root}>
      <AppBar position="static">
        <Toolbar>
          <IconButton edge="start" className={classes.menuButton} color="inherit" aria-label="menu">
            <MenuIcon/>
          </IconButton>
          <Typography variant="h6" className={classes.title}>
            DumbAFCbts
          </Typography>
          {isLoggedIn ?
           <div className={'user-info--wrapper'}>
             <Button className={'user-info--username'} color="inherit">{user.userName}</Button>
             <Button className={'user--logout-button'} color='inherit'
                     onClick={() => dispatch(signOut())}>LOGOUT</Button>
           </div>
                      :
           <Button className={'user--login-button'} color="inherit"
                   onClick={() => dispatch(signInStart())}>LOGIN</Button>
          }
          <div className={'theme-toggle--button'} onClick={props.toggleTheme}>
            <IconButton color="inherit">
              <LightbulbIcon variant={props.themeType}/>
            </IconButton>
          </div>
        </Toolbar>
      </AppBar>
    </div>
  );
};