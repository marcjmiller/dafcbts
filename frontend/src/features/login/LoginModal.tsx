import React from 'react';
import { baseStyles } from '../../resources/theme';
import { useDispatch } from 'react-redux';
import { Button, Modal, TextField, Typography } from '@material-ui/core';
import { resetSignIn, signInSuccess } from '../../store/reducer/slices/authSlice';
import classNames from 'classnames';

interface MyProps {
  display: boolean;
}

const LoginModal: React.FC<MyProps> = (props) => {
  const classes = baseStyles();
  const dispatch = useDispatch();

  return (
    <Modal
      open={props.display}
      onClose={() =>
        dispatch(resetSignIn())
      }
      style={{ top: '50%', left: '50%' }}
    >
      <div className={classNames(classes.modalBody, classes.paper, 'login--modal')}>
        <Typography>LOGIN</Typography>
        <div className={'modal-text'}>
          <TextField id="username-email--input" label="Username or Email" variant="outlined" autoFocus={true}/>
        </div>
        <div className={'modal-text'}>
          <TextField id="password--input" label="Password" type="password" variant="outlined"/>
        </div>
        <Button
          variant="contained"
          color="primary"
          onClick={() =>
            dispatch(signInSuccess({ userName: 'Billy', token: 'is logged in' }))
          }
        >
          SUBMIT
        </Button>
      </div>
    </Modal>
  );
};

export default LoginModal;