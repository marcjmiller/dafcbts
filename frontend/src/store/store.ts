import { configureStore } from '@reduxjs/toolkit';
import { rootReducer } from './reducer';

export const store = configureStore({
  reducer: rootReducer,
});

// TODO: make sure this works, and remove it if it doesn't
//@ts-ignore
if (process.env.NODE_ENV === 'development' && module.hot) {
  //@ts-ignore
  module.hot.accept('./reducer/rootReducer', () => {
    const newRootReducer = require('./reducer/rootReducer').default;
    store.replaceReducer(newRootReducer);
  })
}

export type AppDispatch = typeof store.dispatch;