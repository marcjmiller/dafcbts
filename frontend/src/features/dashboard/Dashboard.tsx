import React, { useEffect, useState } from 'react';
import { useStyles } from '../../resources/theme';
import { useDispatch, useSelector } from 'react-redux';
import { RootState } from '../../resources/types';
import { fetchAllCbts } from '../../store/actions';
import { Button } from '@material-ui/core';

const Dashboard = () => {
  const dispatch = useDispatch();
  const classes = useStyles();
  const [refresh, setRefresh] = useState(true);

  useEffect(() => {
    dispatch(fetchAllCbts());
  }, [refresh]);

  const { cbts } = useSelector(
    (state: RootState) => state.cbts,
  );

  return (
    <div className={classes.dashboardContainer}>
      <div className={classes.dashboard}>
        Cbts fetched: {cbts.length}
        <Button onClick={() => setRefresh(!refresh)} color={'primary'} className={'refetch'}>Re-Fetch</Button>
      </div>
    </div>
  );
};

export default Dashboard;