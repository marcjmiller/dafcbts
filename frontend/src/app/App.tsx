import React from 'react';
import { createMuiTheme, CssBaseline, ThemeProvider } from '@material-ui/core';
import { NavBar } from '../features/dashboard/NavBar';
import { useDispatch, useSelector } from 'react-redux';
import { toggleThemeType } from '../store/reducer/slices/themeSlice';
import { RootState } from '../resources/types';
import { AuthStep } from '../store/reducer/slices/authSlice';
import LoginModal from '../features/login/LoginModal';
import Dashboard from '../features/dashboard/cbtDashboard/CbtDashboard';
import Footer from '../features/dashboard/Footer';
import { baseStyles } from '../resources/theme';

const App: React.FC = () => {
  const dispatch = useDispatch();

  const { themeType } = useSelector(
    (state: RootState) => state.theme,
  );

  const { authStep } = useSelector(
    (state: RootState) => state.auth,
  );

  const muiTheme = createMuiTheme({
    palette: {
      type: themeType,
    },
  });

  const toggleTheme = () => {
    dispatch(toggleThemeType());
  };

  const classes = baseStyles();
  const isLoggingIn = authStep === AuthStep.LOGGING_IN;

  return (
    <ThemeProvider theme={muiTheme}>
      <CssBaseline/>
      <div className={classes.app}>
        <NavBar themeType={themeType} toggleTheme={toggleTheme}/>
        <Dashboard/>
        <Footer/>
        <LoginModal display={isLoggingIn}/>
      </div>
    </ThemeProvider>
  );
};

export default App;