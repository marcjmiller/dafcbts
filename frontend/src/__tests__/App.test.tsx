import React from 'react';
import App from '../app/App';
import { mount } from 'enzyme';
import { NavBar } from '../features/dashboard/NavBar';
import { Provider } from 'react-redux';
import { store } from '../store';
import LoginModal from '../features/login/LoginModal';
import Dashboard from '../features/dashboard/cbtDashboard/CbtDashboard';
import Footer from '../features/dashboard/Footer';

describe('App unit tests', () => {
  const wrapper = mount(
    <Provider store={store}>
      <App/>
    </Provider>,
  );

  it('should contain the NavBar component', () => {
    expect(wrapper.find(NavBar).exists()).toBeTruthy();
  });

  it('should display a login button that displays a login window after clicking', () => {
    expect(wrapper.find('.user--login-button').at(1).text()).toContain('LOGIN');
    expect(wrapper.find('.user--login-button').at(1).simulate('click'));
    wrapper.update();
    expect(wrapper.find(LoginModal).exists()).toBeTruthy();
  });

  it('should render the dashboard', () => {
    expect(wrapper.find(Dashboard).exists()).toBeTruthy();
  });

  it('should render the footer', () => {
    expect(wrapper.find(Footer).exists()).toBeTruthy();
  });

  // TODO: fix snapshot testing
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});


