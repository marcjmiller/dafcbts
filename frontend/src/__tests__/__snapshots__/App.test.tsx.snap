// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App unit tests should match last snapshot also 1`] = `
<Fragment>
  <WithStyles(CssBaseline) />
  <div
    className="App"
  >
    <header
      className="App-header"
    >
      Hello World
    </header>
  </div>
</Fragment>
`;
